src/MAUI/Maui.Samples/SamplePage.xaml.cs

```diff
@@ -112,29 +108,23 @@ private async void LoadSampleData(SampleInfo sampleInfo)
             }
         }
 
-        private void MainPagePropertyChanged(object sender, PropertyChangedEventArgs e)
-        {
-            if (e.PropertyName == "CurrentPage")
-            {
-                TryDispose();
-            }
-        }
-
-        private void NavigatedFromEvent(object sender, NavigatedFromEventArgs e)
-        {
-            TryDispose();
-        }
-
-        private void TryDispose()
+        private void SamplePage_NavigatedFrom(object sender, NavigatedFromEventArgs e)
         {
             // Check that the navigation is backward from the sample and not forward into another page in the sample.
             if (!Application.Current.MainPage.Navigation.NavigationStack.OfType<SamplePage>().Any())
```

Comment:
I believe this will always be false, because current page is still in the NavigationStack when NavigatedFrom event fires. Try setting a breakpoint inside this method to see if it's actually reached.

Reply:
It's making it inside the conditional when I place a breakpoint at the for loop. It's having expected behavior - I tested the ArcGIS token challenge sample which has an additional page. And tested a sample without an additional page.

Reply:
Hmm you're right, I just retested and it works.  False alarm!  I'm not sure what I was doing wrong before, sorry about that.